### PR TITLE
Add summary section to main form

### PR DESCRIPTION
**Pull request by gbochora**
#### Summary
This PR adds the `Summary` section to the new run details page. Added a column to the `IR_Incident` table to store the last modified date
`Summary` section includes:
- Title and last edited info
- `Copy link` to jump to the `Summary` section, it appears on hover
- Markdown edit to edit the `Summary`

Link to visit new run details page: `http://localhost:8065/playbooks/run_details/[run_id]/`

image

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-44714

#### Checklist

~~- [ ] Telemetry updated~~
~~- [ ] Gated by experimental feature flag~~
~~- [ ] Unit tests updated~~

### CODE REVIEW

File: webapp/src/components/backstage/playbook_runs/playbook_run/summary.tsx [hunk 1]
```diff
@@ -1,13 +1,91 @@
...
+import {updatePlaybookRunDescription} from 'src/client';
+import {Timestamp} from 'src/webapp_globals';
+
+import {AnchorLinkTitle} from './playbook_run';
+
+interface Props {
+    playbookRun: PlaybookRun;
+}
+
+const EDIT_TIME = {
```
Comment (trilopin) [inline webapp/src/components/backstage/playbook_runs/playbook_run/summary.tsx:17]: is this used on more places? it doesn't seem to be too summary-related.

File: webapp/src/components/backstage/playbook_runs/playbook_run/summary.tsx [hunk 2]
```diff
@@ -1,13 +1,91 @@
...
+`;
+
 const Container = styled.div`
     display: flex;
-    flex-direction: row;
-`;
\ No newline at end of file
+    flex-direction: column;
+    max-width: 720px;
+    min-width: 614px;
```
Comment (trilopin) [inline webapp/src/components/backstage/playbook_runs/playbook_run/summary.tsx:90]: are you sure about these widths? shouldn't we do this in the global container and work with all section containers at 100%?

File: webapp/src/components/backstage/playbook_runs/playbook_run/summary.tsx [hunk 3]
```diff
@@ -1,13 +1,91 @@
...
+                    title={title}
+                    id='summary'
+                />
+                {playbookRun.summary_modified_at > 0 && modifiedAtMessage}
+            </Header>
+            <MarkdownEdit
+                placeholder={formatMessage({defaultMessage: 'Add a run summary'})}
+                value={playbookRun.summary}
+                onSave={(value) => {
+                    updatePlaybookRunDescription(playbookRun.id, value);
```
Comment (trilopin) [inline webapp/src/components/backstage/playbook_runs/playbook_run/summary.tsx:61]: it's weird that we have a mix of description and summary naming.

File: webapp/src/components/backstage/playbook_runs/playbook_run/playbook_run.tsx [hunk 4]
```diff
@@ -96,4 +99,43 @@ const Body = styled.main`
...
+`;
+
+interface AnchorLinkTitleProps {
+    title: string;
+    id: string;
+
+}
+
+export const AnchorLinkTitle = (props: AnchorLinkTitleProps) => {
+    const {url} = useRouteMatch();
```
Comment (trilopin) [inline webapp/src/components/backstage/playbook_runs/playbook_run/playbook_run.tsx:111]: it feels a bit weird to import the main component from the subcomponents.  wouldn't it be nicer if we create a file with this kind of tiny shared component?

**Review by trilopin — APPROVED** [review]
Nice job!

All my comments are optional

**Comment by trilopin** [comment]
Hey @gbochora, I noticed that in the design the edit icon and the last edited date appear when doing hover on the whole section.

However in the implemented component:
- hover only works when the mouse is over the textarea (not the title) 
- the last edit is always visible, not on hover

File: webapp/src/components/backstage/playbook_runs/playbook_run/summary.tsx (hunk 2 above)
Comment (gbochora) [reply webapp/src/components/backstage/playbook_runs/playbook_run/summary.tsx:90]: It's a temporary solution. I'll move it to the `playbook_run`

File: webapp/src/components/backstage/playbook_runs/playbook_run/summary.tsx (hunk 3 above)
Comment (gbochora) [reply webapp/src/components/backstage/playbook_runs/playbook_run/summary.tsx:61]: Yes, it was weird to me too. I'll make a note to fix this in the future.

File: webapp/src/components/backstage/playbook_runs/playbook_run/playbook_run.tsx (hunk 4 above)
Comment (gbochora) [reply webapp/src/components/backstage/playbook_runs/playbook_run/playbook_run.tsx:111]: Very good point! I'll move it to the shared file.

**Review by calebroseland — APPROVED** [review]

**Pull request merged**